libsolidity/ast/Types.h

```diff
@@ -1050,6 +1050,13 @@ class FunctionType: public Type
 		m_bound(_bound),
 		m_declaration(_declaration)
 	{
+		if (m_parameterNames.empty() && !m_parameterTypes.empty())
+			m_parameterNames = strings(m_parameterTypes.size(), "");
```

Comment:
I would prefer to do this filling in the special-purpose constructor above only ("`Function type constructor to be used for a plain type (not derived from a declaration).`") and add the assertion here.

Reply:
I am now working on that. So far it requires fixes all over the code because both c'tors are used